src/Modules/UngradedModules/Tensor.jl

```diff
@@ -115,8 +115,12 @@ julia> t((M[1], M[2]))
 x*y*e[1] \otimes e[1]
 ```
 """
-function tensor_product(G::ModuleFP...; task::Symbol = :none)
-  F, mF = tensor_product([ambient_free_module(x) for x = G]..., task = :map)
+function tensor_product(G::ModuleFP...; 
+    ambient_tensor_product::FreeMod=tensor_product([ambient_free_module(x) for x = G]..., task = :none),
```

Comment:
```suggestion
    ambient_tensor_product::FreeMod=tensor_product((ambient_free_module(x) for x in G)..., task = :none),
```
this saves allocating a vector that gets splatted immediately anyway